File: app/src/main/java/com/example/javakotlin/javakotlinoverview/java/MobileDepartment.java
package com.example.javakotlin.javakotlinoverview.java;

interface MobileDepartment {
    void getIDE(String ide);
}

class Bansi implements MobileDepartment {

    public void getIDE(String ide) {
        System.out.println("Bansi is working in " + ide);
    }
}

class ABC implements MobileDepartment {
    public void getIDE(String ide) {
        System.out.println("ABC is working in " + ide);
    }
}

class InterfaceDemo {
    public static void main(String[] args) {
        Bansi b = new Bansi();
        b.getIDE("Android Studio");

        ABC abc = new ABC();
        abc.getIDE("XCode");
    }
}

File: app/src/main/java/com/example/javakotlin/javakotlinoverview/kotlin/MobileDepartment.kt
package com.example.javakotlin.javakotlinoverview.kotlin

interface MobileDepartment {
    fun getIDE() : String
    fun message(){
        println("This is mobile department")
    }
}

class Bansi : MobileDepartment {
    override fun getIDE() = "Android Studio"
}

class ABC : MobileDepartment {
    override fun getIDE() = "Xcode"
}

fun main(args: Array<String>) {
    val b = Bansi()
    println("Working in ${b.getIDE()}")
    b.message()

    val abc = ABC()
    println("Working in ${abc.getIDE()}")
    abc.message()
}